catalog/core/catalog-core-api/src/main/java/ddf/catalog/source/Source.java

```diff
@@ -34,24 +34,42 @@
    *
    * <p><b>This is expected to be an expensive operation, possibly involving network I/O.</b>
    * Typically only {@link ddf.catalog.CatalogFramework} implementations will call this and only
-   * periodically.
+   * periodically. {@link
+   * ddf.catalog.CatalogFramework#getSourceInfo(ddf.catalog.operation.SourceInfoRequest)} should be
+   * used as a non-blocking alternative to {@link #isAvailable()}.
    *
    * @return true - if the site is available (up), false - if the site is unavailable (down)
-   * @see ddf.catalog.util.SourcePoller
    */
   public boolean isAvailable();
 
   /**
    * This method is the same as {@link #isAvailable()} but allows a caller to provide a {@link
-   * SourceMonitor} callback object which is meant to be used as a way for the {@link Source} to
-   * dynamically contact the caller with its availability. The intent is to return a boolean as to
+   * SourceMonitor} which is meant to be used as a way for the {@link Source} to dynamically update
+   * the caller when its availability changes. The intent is to return a {@code boolean} as to
    * whether the {@link Source} is available at this very moment and also to use the {@link
-   * SourceMonitor} object to update the caller of this method if this Source's availability changes
-   * later in the future.
+   * SourceMonitor} to update the caller if the availability of the {@link Source} is checked
```

Comment:
✏️Should we keep _update the caller of this method if this Source's availability changes later in the future_ here? I think it conveys the intent a little better.

Reply:
I think that "update the caller of this method if this Source's availability changes later in the future" is misleading because it makes it seem like the caller will be updated immediately when the availability changes when none of the current implementations do that. Current DDF implementations only update the caller when `isAvailable()` is called and the availability has changed--there is no update if the availability changes between `isAvailable()` calls.

Reply:
The CatalogProvider do support this interface properly. This was changed when the Solr factory was redesigned.

As for the other ones, Isn't it because none of the implementation took time to actually implement this method correctly?

Reply:
Actually I think CSW does it:
https://github.com/codice/ddf/blob/master/catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswSource.java#L1701-L1722

Reply:
I think the point is not to assume because some didn't implement this listener mechanism properly that all didn't.

Reply:
:+1: Removed